### PR TITLE
Add some integration tests

### PR DESCRIPTION
**Pull request by nalind**
This adds some integration tests for exercising the higher-level APIs by way of using the CLI helper.  We add read-only store support to the "vfs" driver to let us always test that option, too.

### CODE REVIEW

File: tests/stores.bats [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	storage --graph ${TESTDIR}/ro-root --run ${TESTDIR}/ro-runroot unmount $lowerlayer
+
+	# Create an image using this second layer.
+	run storage --graph ${TESTDIR}/ro-root --run ${TESTDIR}/ro-runroot --debug=false create-image $midlayer
+        [ "$status" -eq 0 ]
+        [ "$output" != "" ]
+        image=${output%%  *}
+
+	# Create a third layer based on the second one.
+	run storage --storage-opt ${STORAGE_DRIVER}.imagestore=${TESTDIR}/ro-root create-layer "$midlayer"
```
Comment (rhatdan) [inline tests/stores.bats]: Shouldn't you check status on this run?

**Comment by TomSweeneyRedHat** [comment]
Other than Dan's question and just a generic can you add more status checks?  LGTM.

File: tests/stores.bats (hunk 1 above)
Comment (nalind) [reply tests/stores.bats]: Whoops, that's a leftover from when it was being worked out.  There's no need to call it that first time.

**Comment by nalind** [comment]
I think every call to the `run` helper is followed by a check on the `$status` that it would have set.  Can you point me at any that are missing?

**Comment by TomSweeneyRedHat** [comment]
I didn't spot any 'run' calls without checks, just a number of places that didn't have run and did some call without a check.  For instance tests/apply-diff.bats you do "checkchanges" and "checkdiffs" at the end.  Can those be checked?  They seem to me like they could, but your call.

I wouldn't hold due to those, just something to consider for later revisions.

**Comment by nalind** [comment]
Ah, there's no need to do that.  Functions like `checkchanges` and `checkdiffs` check `$status` after using `run`, so they fail if that comparison fails.  Those functions aren't called using `run`, so if they fail, the tests or other functions that called them will fail.

Thanks for catching places where `$status` was being checked after it wouldn't have been set; those were also leftovers from when earlier iterations, and they should be gone now.

**Comment by nalind** [comment]
Updated to skip tests that require 'tar' when it isn't available.

**Comment by nalind** [comment]
[(5) [containers/storage] Add tests for layer diff contents](https://trello.com/c/VV1Q408f/656-5-containers-storage-add-tests-for-layer-diff-contents)

**Comment by nalind** [comment]
Updated the test harness to test with all available drivers.  It'll take a bit longer to run, though.

**Comment by nalind** [comment]
Undoing that last change, for now.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**